packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -0,0 +1,174 @@
+    super.initState();
+    _controller = new AnimationController(duration: _kExpand, vsync: this);
+    _easeOutAnimation = new CurvedAnimation(parent: _controller, curve: Curves.easeOut);
+    _easeInAnimation = new CurvedAnimation(parent: _controller, curve: Curves.easeIn);
+    _borderColor = new ColorTween(begin: Colors.transparent);
+    _headerColor = new ColorTween();
+    _iconColor = new ColorTween();
+    _iconTurns = new Tween<double>(begin: 0.0, end: 0.5).animate(_easeInAnimation);
+    _backgroundColor = new ColorTween();
+
+    _isExpanded = PageStorage.of(context)?.readState(context) ?? false;
+    if (_isExpanded)
+      _controller.value = 1.0;
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void _handleOnTap() {
+    setState(() {
+      _isExpanded = !_isExpanded;
+      if (_isExpanded)
+        _controller.forward();
+      else
+        _controller.reverse();
+      PageStorage.of(context)?.writeState(context, _isExpanded);
+    });
+    if (widget.onOpenChanged != null)
+      widget.onOpenChanged(_isExpanded);
+  }
+
+  Widget _buildChildren(BuildContext context, Widget child) {
+    return new Container(
+      decoration: new BoxDecoration(
+        color: _backgroundColor.evaluate(_easeOutAnimation),
+        border: new Border(
+          top: new BorderSide(color: _borderColor.evaluate(_easeOutAnimation)),
+          bottom: new BorderSide(color: _borderColor.evaluate(_easeOutAnimation))
+        )
+      ),
+      child: new Column(
+        children: <Widget>[
+          IconTheme.merge(
+            data: new IconThemeData(color: _iconColor.evaluate(_easeInAnimation)),
+            child: new ListTile(
+              onTap: _handleOnTap,
+              leading: widget.leading,
+              title: new DefaultTextStyle(
+                style: Theme.of(context).textTheme.subhead.copyWith(color: _headerColor.evaluate(_easeInAnimation)),
+                child: widget.title
+              ),
+              trailing: new RotationTransition(
+                turns: _iconTurns,
+                child: const Icon(Icons.expand_more)
+              )
+            )
+          ),
+          new ClipRect(
+            child: new Align(
+              heightFactor: _easeInAnimation.value,
+              child: new Column(children: widget.children)
+            )
+          )
+        ]
+      )
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    _borderColor.end = theme.dividerColor;
+    _headerColor
+      ..begin = theme.textTheme.subhead.color
+      ..end = theme.accentColor;
+    _iconColor
+      ..begin = theme.unselectedWidgetColor
+      ..end = theme.accentColor;
+    _backgroundColor
+      ..begin = Colors.transparent
+      ..end = widget.backgroundColor ?? Colors.transparent;
+
+    return new AnimatedBuilder(
+      animation: _controller.view,
+      builder: _buildChildren
+    );
```

Comment:
I'd toss the `new Column(children: widget.children)` in here as the `child` parameter so the `Column` doesn't need to rebuild each tick of the animation.